design/architecture/api.md

```diff
@@ -19,4 +19,132 @@ and rules on how files and folders are structured and named. Taken
 together, this API would allow other software like R packages to be
 built to interact with the backend to automate tasks done by the users.
 
-TODO: File tree structure design with API endpoints.
+---
+API security 
+---
+
+Adding a security layer to an API is crucial for ensuring the confidentiality,
+integrity, and availability of the data and systems that use the API. APIs are
+often used to connect different systems and applications, and they provide a way
+for external parties to access and interact with data. Because of this, they can
+be an attractive target for cyber attackers looking to gain unauthorized access,
+steal sensitive information, or disrupt service availability. To protect against
+these threats, it is important to implement a robust security strategy for the
+API.
+
+
+### Approach 1: Basic Authentication
+
+Here is an example for setting up with basic user name password for API request
+
+`````
+class SecureView(APIView):
+    """ Setup secure view of the API with basic authentication """
+    authentication_classes = (BasicAuthentication,)
+    permission_classes = (IsAuthenticated,)
+    
+`````
```

Comment:
At this stage we don't need actual code, more a higher level description (maybe including the hypothetical methods or functions used) of what each approach does and your recommended approach.